ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java

```diff
@@ -167,4 +168,22 @@ public void start(final QueryId queryId) {
       throw new IllegalArgumentException("query: " + queryId + " not added to runtime");
     }
   }
+
+  @Override
+  public void restartStreamsRuntime() {
+    final KafkaStreamsNamedTopologyWrapper kafkaStreamsNamedTopologyWrapper = kafkaStreamsBuilder
+        .buildNamedTopologyWrapper(streamsProperties);
+    kafkaStreams = kafkaStreamsNamedTopologyWrapper;
+    for (final BinPackedPersistentQueryMetadataImpl binPackedPersistentQueryMetadata
+        : collocatedQueries.values()) {
+      kafkaStreamsNamedTopologyWrapper.addNamedTopology(
+          binPackedPersistentQueryMetadata.getTopology()
+      );
+    }
+    kafkaStreams.close();
+    kafkaStreams.cleanUp();
```

Comment:
We should avoid calling cleanup to keep this lightweight, otherwise each config change will restore everything from scratch 😬

Reply:
good call